src/executePrettier/executePrettierOnBufferRange.js

```diff
@@ -15,18 +19,30 @@ const executePrettier = (editor: TextEditor, text: string) =>
 const executePrettierEslint = (editor: TextEditor, text: string) =>
   allowUnsafeNewFunction(() => prettierEslint(buildPrettierEslintOptions(editor, text)));
 
-const executePrettierOrPrettierEslint = (editor: TextEditor, text: string) => {
+const executePrettierStylelint = (editor: TextEditor, text: string) =>
+  prettierStylelint.format(buildPrettierStylelintOptions(editor, text));
+
+const executePrettierOrIntegration = async (editor: TextEditor, text: string) => {
   try {
-    return shouldUseEslint() ? executePrettierEslint(editor, text) : executePrettier(editor, text);
+    let formatted;
+    if (shouldUseStylelint() && isCurrentScopeCssScope(editor)) {
+      formatted = await executePrettierStylelint(editor, text);
+    } else if (shouldUseEslint()) {
+      formatted = executePrettierEslint(editor, text);
+    } else {
+      formatted = executePrettier(editor, text);
+    }
+
+    return formatted;
   } catch (error) {
     return error;
   }
 };
 
-const executePrettierOnBufferRange = (editor: TextEditor, bufferRange: Range) => {
+const executePrettierOnBufferRange = async (editor: TextEditor, bufferRange: Range) => {
   const cursorPositionPriorToFormat = editor.getCursorScreenPosition();
   const textToTransform = editor.getTextInBufferRange(bufferRange);
-  const transformed = executePrettierOrPrettierEslint(editor, textToTransform);
+  const transformed = await executePrettierOrIntegration(editor, textToTransform);
```

Comment:
Have you verified that with babel-polyfill, Atom will properly wait before continuing on? I used to have a problem using `async` in this repo because Atom wouldn't wait, but I never thought to try using babel-polyfill.

Reply:
It will not wait, async/await will still work as Promise so we are making the executePrettierOnBufferRange function asynchronous, which should cause no issues at all, as soon as the promise is resolved runLinter will called normally.

I have been using prettier-atom with this changes for a couple of days and everything has been working normally.

Reply:
Roger that!